src/dos/drive_fat.cpp

```diff
@@ -736,12 +738,12 @@ fatDrive::fatDrive(const char *sysFilename,
 		imgDTAPtr = RealMake(imgDTASeg, 0);
 		imgDTA    = new DOS_DTA(imgDTAPtr);
 	}
-
-	diskfile = fopen_wrap(sysFilename, "rb+");
+	diskfile = fopen_wrap_ro_fallback(sysFilename, roflag);
 	if (!diskfile) {
 		created_successfully = false;
 		return;
-	}
+	} else
+		readonly = roflag;
```

Comment:
And can then do something like:

```
	FILE *diskfile = fopen_wrap_ro_fallback(sysFilename, readonly);
	created_successfully = (diskfile != nullptr);
	if (!created_successfully)
		return;
```

Then `fopen_wrap_ro_fallback()` will adjust `readonly` for us, if it does need to fallback to `ro`.

Reply:
(this is minor; I can't see anything else in the PR)

Reply:
@kcgen I can confirm the simplified code works. Thanks for the suggestion.